replication/binlogsyncer.go

```diff
@@ -169,11 +168,10 @@ type BinlogSyncer struct {
 // NewBinlogSyncer creates the BinlogSyncer with the given configuration.
 func NewBinlogSyncer(cfg BinlogSyncerConfig) *BinlogSyncer {
 	if cfg.Logger == nil {
-		streamHandler, _ := log.NewStreamHandler(os.Stdout)
-		cfg.Logger = log.NewDefault(streamHandler)
+		cfg.Logger = slog.Default()
 	}
 	if cfg.ServerID == 0 {
-		cfg.Logger.Fatal("can't use 0 as the server ID")
+		panic("can't use 0 as the server ID")
```

Comment:
panic may send output to a different place than `cfg.Logger`, I prefer

```suggestion
		cfg.Logger.Error("can't use 0 as the server ID, will panic")
		panic("can't use 0 as the server ID")
```